### PR TITLE
Enable mkl docs

### LINKED ISSUE #750: ENABLE_MKL not described in manual and README

### PR DESCRIPTION
**Pull request by PDoakORNL**
Adding documentation for ENABLE_MKL.
Closes #750

### CODE REVIEW

File: manual/installation.tex [hunk 1]
```diff
@@ -196,71 +203,89 @@ \subsection{Configuration options}
...
+QMC_COMPLEX           Build the complex (general twist/k-point) version (1:yes, 0:no)
+QMC_MIXED_PRECISION   Build the mixed precision (mixing double/float) version
+                      (1:yes (GPU default), 0:no (CPU default)).
+                      The CPU support is experimental.
+                      Use float and double for base and full precision.
+                      The GPU support is quite mature.
+                      Use always double for host side base and full precision
+                      and use float and double for CUDA base and full precision.
+ENABLE_MKL            Enable Intel MKL libraries.
+MKL_ROOT              Path to MKL libraries (only necessary for non intel compilers
```
Comment (ye-luo) [inline manual/installation.tex]: ENABLE_MKL and MKL_ROOT are not key build options. Please list them under addition build options

File: manual/installation.tex [hunk 2]
```diff
@@ -392,10 +422,11 @@ \subsection{Using vendor optimized numerical libraries, e.g. Intel MKL}
 \subsubsection{Using Intel MKL with non-Intel compilers}
 
 To use Intel MKL with, e.g. an MPICH wrapped gcc:
+\verbatimfont{\footnotesize}
 \begin{verbatim}
 cmake \
  -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx \
- -DBLA_VENDOR=Intel10_64lp_seq -DCMAKE_PREFIX_PATH=$MKLROOT/lib \
+ -DBLA_VENDOR=Intel10_64lp_seq -DENABLE_MKL -DMKL_ROOT=$MKLROOT/lib \
```
Comment (ye-luo) [inline manual/installation.tex]: I don't think cmake accepts -DENABLE_MKL without a value.
Is just -D DENABLE_MKL=1 sufficient without DBLA_VENDOR and MKL_ROOT?

**Comment by prckent** [comment]
OK to test

File: manual/installation.tex (hunk 2 above)
Comment (PDoakORNL) [reply manual/installation.tex]: If a MKL_HOME, MKLROOT, or MKL_ROOT environment variable is set, then yes it is.

**Comment by PDoakORNL** [comment]
I believe I've fixed the itemize environments. I don't know why I broke them. Spasmodic pasting is what it looks like.

File: manual/installation.tex [hunk 3]
```diff
@@ -331,9 +359,10 @@ \subsection{Example configure and build}
 \end{verbatim}
 
 \item Move to build directory, run cmake and make
+\verbatimfont{\footnotesize}
 \begin{verbatim}
 cd build
-cmake -D CMAKE_BUILD_TYPE=Release ..
+cmake -D CMAKE_BUILD_TYPE=Release -DENABLE_MKL=1 ..
```
Comment (ye-luo) [inline manual/installation.tex]: In this case, if MKL_HOME is specified. No need to put it on cmake.

File: manual/installation.tex [hunk 4]
```diff
@@ -196,71 +203,89 @@ \subsection{Configuration options}
...
+\normalsize
 \item Additional QMCPACK build options
+\verbatimfont{\footnotesize}
 \begin{verbatim}
 QMC_INCLUDE         Add extra include paths
 QMC_EXTRA_LIBS      Add extra link libraries
 QMC_BUILD_STATIC    Add -static flags to build
 QMC_DATA            Specify data directory for QMCPACK performance and integration tests
 QE_BIN              Location of Quantum Espresso binaries including pw2qmcpack.x
+ENABLE_MKL          Enable Intel MKL libraries.
```
Comment (ye-luo) [inline manual/installation.tex]: Please do as other libraries. Not mixed with QMC_XXX.
Also mention turn on by default when intel compiler is used.

File: manual/installation.tex [hunk 5]
```diff
@@ -292,13 +318,14 @@ \subsection{Role of QMC\_DATA}
 QMC_DATA/NiO/NiO-fcc-supertwist111-supershift000-S128.h5
 QMC_DATA/NiO/NiO-fcc-supertwist111-supershift000-S256.h5
 QMC_DATA/NiO/NiO_afm_fcidump.h5
-QMC_DATA/NiO/NiO_afm_wfn.dat 
-QMC_DATA/NiO/NiO_nm_choldump.h5   
+QMC_DATA/NiO/NiO_afm_wfn.dat
+QMC_DATA/NiO/NiO_nm_choldump.h5
 \end{verbatim}
-
+\normalsize
```
Comment (ye-luo) [inline manual/installation.tex]: I sometimes see normalsize, sometimse don't. Could you explain why?

File: manual/installation.tex [hunk 6]
```diff
@@ -392,10 +422,11 @@ \subsection{Using vendor optimized numerical libraries, e.g. Intel MKL}
 \subsubsection{Using Intel MKL with non-Intel compilers}
 
 To use Intel MKL with, e.g. an MPICH wrapped gcc:
+\verbatimfont{\footnotesize}
 \begin{verbatim}
 cmake \
  -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx \
- -DBLA_VENDOR=Intel10_64lp_seq -DCMAKE_PREFIX_PATH=$MKLROOT/lib \
+ -DBLA_VENDOR=Intel10_64lp_seq -DENABLE_MKL=1 -DMKL_ROOT=$MKLROOT/lib \
```
Comment (ye-luo) [inline manual/installation.tex]: I think if ENABLE_MKL=1 is enabled, BLA_VENDOR is skipped completely.

File: manual/installation.tex [hunk 7]
```diff
@@ -175,10 +179,13 @@ \subsection{Environment variables}
 A number of environmental variables affect the build.  In particular
 they can control the default paths for libraries, the default
 compilers, etc.  The list of environmental variables is given below:
+\verbatimfont{\footnotesize}%
 \begin{verbatim}
 CXX              C++ compiler
 CC               C Compiler
 MKL_HOME         Path for MKL
+MKL_ROOT
```
Comment (ye-luo) [inline manual/installation.tex]: Sorry. I was giving the wrong info.
Please list only MKL_HOME for environment variable.
When we tell users, just give them 1 but our setting can be smarter internally.

File: manual/installation.tex (hunk 5 above)
Comment (PDoakORNL) [reply manual/installation.tex]: You shouldn't see it at all.  Will fix.

File: manual/installation.tex (hunk 6 above)
Comment (PDoakORNL) [reply manual/installation.tex]: Correct should be deleted

File: manual/installation.tex (hunk 7 above)
Comment (PDoakORNL) [reply manual/installation.tex]: Good that will be less confusing

**Comment by ye-luo** [comment]
A question.
We have MKL_HOME historically. With the improved MKL setting, MKL_ROOT was introduced and used heavily in FindMKL script.
Can we reduce them to one? I prefer MKL_HOME since this is used historically.
When MKL_HOME cmake variable is not defined, we can still look for MKL_ROOT, MKLROOT, ENV{MKLROOT}, ENV{MKL_HOME}

**Comment by PDoakORNL** [comment]
I used MKL_ROOT because it's more often used in cmake FindXXX.cmake includes. Since I've worked on several of them I of course want them to be as consistent as possible inside.
While the Findxxx.cmake's are far not 100% consistent they seem more consistent that different vendors chose environment variables.

**Comment by ye-luo** [comment]
VASP has no configure step and users directly define MKL setting in Makefile.
QE directly inject MKL when Intel compiler is used. If mixing GNU and MKL is needed, modify make.inc.
Gromacs needs -DMKL_INCLUDE_DIR and -DMKL_LIBRARIES explicitly.
LAMMPS has the same setting as Gromacs.
None of them has advanced script like us.

**Comment by ye-luo** [comment]
I'm OK to have either MKL_HOME or MKL_ROOT but not MKL_HOME in the environment with MKL_ROOT in the cmake.
however, Intel defines MKLROOT which is too close to MKL_ROOT.
We can leave the change to a separate PR but I preferred to get this settled before this is exposed to users (release).

**Comment by PDoakORNL** [comment]
It just annoys me. I vote remove MKL_HOME from the example and stay with MKL_ROOT. I think we should be like HDF5 and BOOST who have better find scripts (I think) than than FFTW and LIBXML2.  

Like us findBOOST supports BOOST_HOME, BOOST_ROOT, and BOOSTROOT

I prefer to define everything in the cmake command. But then I'm building many versions and like to be able to capture shell history and not worry about too many enviroment variables. This is not the user situation.

**Comment by prckent** [comment]
My vote is also now for the consistency argument - MKL_ROOT as the primary spec, as per HDF5 and BOOST. We can silently support MKL_HOME for backwards compatibility, but lets advertise MKL_ROOT as the preferred way.

**Pull request merged**